src/software/estop/uart_communication.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <boost/function.hpp>
+
+/*
+ * A boost::asio wrapper class that is used to synchronously communicate with a serial
+ * device via UART
+ */
+
+class UartCommunication
+{
+    using io_service      = boost::asio::io_service;
+    using serial_port_ptr = std::shared_ptr<boost::asio::serial_port>;
+
+   public:
+    enum flush_type
+    {
+        flush_receive = TCIFLUSH,  // flushes data received but not read
+        flush_send    = TCOFLUSH,  // flushes data written but not transmitted
+        flush_both    = TCIOFLUSH  // flushes both data received but not read and data
+                                   // written but not transmitted
+    };
+
+    UartCommunication();
+
+    /**
+     * Attempts to open a serial connection with the given device port
+     * setting used: No flow control, No parity, 1 stop bit
+     *
+     * @param ioService boost asio construct for managing io operations
+     * @param baudRate the desired baud rate of the connection
+     * @param port the serial port that we want to communicate with
+     * @return true upon success, false otherwise
+     */
+    virtual bool openPort(io_service &ioService, int baudRate, std::string port);
```

Comment:
Sorry I missed this during the first review, but we should be using the [RAII pattern](https://en.cppreference.com/w/cpp/language/raii) when a class needs to own a resource like this. A class that owns a resource should acquire the resource in the constructor, and unacquire the resource in the destructor. You can do this eaisly by making `openPort` and `closePort` private and calling them in the constructor/destructor, respectively.

Reply:
changed to use RAII pattern